packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -207,52 +203,216 @@ function useMediaUploadQueue() {
+
+      startMuting({ id });
+
+      try {
+        const newFile = await stripAudioFromVideo(file);
+
+        if (!isMounted.current) {
+          return;
+        }
+
+        additionalData.web_stories_is_muted = true;
+        finishMuting({ id, file: newFile, additionalData });
+      } catch (error) {
+        // Cancel uploading if there were any errors.
+        cancelUploading({ id, error });
+
+        trackError('upload_media', error?.message);
+      } finally {
+        currentTranscodingItem.current = null;
+      }
+    },
+    [startMuting, finishMuting, stripAudioFromVideo, cancelUploading]
+  );
+
+  const optimizeVideoItem = useCallback(
+    async (item) => {
+      const { id, file, additionalData = {} } = item;
+
+      startTranscoding({ id });
+
+      try {
+        const newFile = await transcodeVideo(file);
+
+        if (!isMounted.current) {
+          return;
+        }
+
+        additionalData.web_stories_media_source = 'video-optimization';
+        finishTranscoding({ id, file: newFile, additionalData });
+      } catch (error) {
+        // Cancel uploading if there were any errors.
+        cancelUploading({ id, error });
+
+        trackError('upload_media', error?.message);
+      } finally {
+        currentTranscodingItem.current = null;
+      }
+    },
+    [startTranscoding, finishTranscoding, transcodeVideo, cancelUploading]
+  );
+
+  const uploadItem = useCallback(
+    async (item) => {
+      const { id, file, resource, additionalData = {} } = item;
+      let { posterFile } = item;
+
+      let newResource;
+
+      const posterFileName = getFileName(file) + '-poster.jpeg';
+
+      startUploading({ id });
+
+      // If we don't have a poster yet (e.g. after converting a GIF),
+      // try to generate one now.
+      if (!resource.poster && !posterFile) {
+        try {
+          posterFile = await getFirstFrameOfVideo(file);
+        } catch {
+          // Not interested in errors here.
+        }
+      }
+
+      trackEvent('upload_media', {
+        file_size: file?.size,
+        file_type: file?.type,
+      });
+
+      //eslint-disable-next-line @wordpress/no-unused-vars-before-return
+      const trackTiming = getTimeTracker('load_upload_media');
+
+      try {
+        // The newly uploaded file won't have a poster yet.
+        // However, we'll likely still have one on file.
+        // Add it back so we're never without one.
+        // The final poster will be uploaded later by uploadVideoPoster().
+        newResource = await uploadFile(file, additionalData);
+
+        if (!isMounted.current) {
+          return;
+        }
+      } catch (error) {
+        // Cancel uploading if there were any errors.
+        cancelUploading({ id, error });
+
+        trackError('upload_media', error?.message);
+      } finally {
+        trackTiming();
+      }
+
       try {
         const { poster, posterId } = await uploadVideoPoster(
```

Comment:
Before calling this, you should check if `posterFile` is not null. Like in the case of image uploads. 

This makes it exit quicker. See 

https://github.com/google/web-stories-wp/blob/ecdc7c487b3a75b86ddb022700659818713786dd/packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js#L467

Reply:
As per my comment above, I think I'm going to refactor the whole `uploadItem` function anyway to separate image and video upload logic here. It's quite confusing.

Reply:
Reworked that no a little bit; resolving.